typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -696,6 +696,11 @@
                  (cl->* ((-lst a) Univ . -> . (-opt -Index))
                         ((-lst a) b (-> a b Univ)
                            . -> . (-opt -Index))))]
+[index-where (-poly (a b)
```

Comment:
looks like we don't need `b` in the binder

Reply:
right, thanks -- dropped